compiler/modules/nback/scaffold.bal

```diff
@@ -428,14 +429,13 @@ function addStringDefn(llvm:Context context, llvm:Module mod, int defnIndex, str
                                       [llvm:constInt(LLVM_INT, TAG_STRING | <int>variant)]);
 }
 
-function addDecimalDefn(llvm:Context context, llvm:Module mod, int defnIndex, string str) returns llvm:ConstPointerValue {
-    byte[] bytes = str.toBytes();
-    bytes.push(0);
-    llvm:ConstValue val = context.constString(bytes);
-    llvm:Type ty = llvm:arrayType("i8", bytes.length());
-    llvm:ConstPointerValue ptr = mod.addGlobal(ty,
+function addDecimalDefn(llvm:Context context, llvm:Module mod, int defnIndex, decimal val) returns llvm:ConstPointerValue {
+    var [leastSignificantVal, mostSignificantVal] = lib:toLeDpd(val);
+    llvm:ConstValue lestSignificant = llvm:constInt("i64", leastSignificantVal);
+    llvm:ConstValue mostSignificant = llvm:constInt("i64", mostSignificantVal);
+    llvm:ConstPointerValue ptr = mod.addGlobal(llvm:arrayType("i64", 2),
```

Comment:
You need to chnage LLVM_DECIMAL_CONST to match changes to balrt.h.

Reply:
fixed